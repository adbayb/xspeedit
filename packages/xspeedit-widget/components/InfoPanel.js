import React from "react";

const InfoPanel = () => {
	return (
		<section>
			<h1>Préambule:</h1>
			<p>
				{
					"XspeedIt est une société d'import / export ayant robotisé toute sa chaîne d'emballage de colis."
				}
			</p>
			<h1>But:</h1>
			<p>
				{
					"La société souhaite trouver un algorithme permettant à ses robots d'optimiser le nombre de cartons d'emballage utilisés. Cette application essaye donc de répondre à cette problématique."
				}
			</p>
			<h1>Propriétés des cartons et articles:</h1>
			<p>
				{
					"Les articles à emballer sont de taille variable, représentée par un entier compris entre 1 et 9. Chaque carton a une capacité de contenance de 10. Ainsi, un carton peut par exemple contenir un article de taille 3, un article de taille 1, et un article de taille 6."
				}
			</p>
			<style jsx>{`
				section {
					padding: 0 15px;
					overflow-y: scroll;
				}

				h1 {
					margin: 30px 0 20px 0;
					color: white;
				}
			`}</style>
		</section>
	);
};

export default InfoPanel;
