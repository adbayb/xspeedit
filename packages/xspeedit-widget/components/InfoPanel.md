**Preview:**

```js
<InfoPanel />;
```

**Example:**

```jsx static
import React from "react";

const Example = () => {
	return (
		<InfoPanel />
	);
};
```
