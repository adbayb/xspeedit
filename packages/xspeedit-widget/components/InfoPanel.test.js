import React from "react";
import InfoPanel from "./InfoPanel.js";

describe("InfoPanel", () => {
	it("should render correctly", () => {
		// GIVEN: notre composant:
		const tree = shallow(<InfoPanel />);
		// THEN: il doit s'afficher correctement en matchant son snapshot:
		expect(tree).toMatchSnapshot();
	});
});
