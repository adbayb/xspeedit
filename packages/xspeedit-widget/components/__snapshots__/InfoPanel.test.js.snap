// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel should render correctly 1`] = `
<section
  className="jsx-3052590751"
>
  <h1
    className="jsx-3052590751"
  >
    Préambule:
  </h1>
  <p
    className="jsx-3052590751"
  >
    XspeedIt est une société d'import / export ayant robotisé toute sa chaîne d'emballage de colis.
  </p>
  <h1
    className="jsx-3052590751"
  >
    But:
  </h1>
  <p
    className="jsx-3052590751"
  >
    La société souhaite trouver un algorithme permettant à ses robots d'optimiser le nombre de cartons d'emballage utilisés. Cette application essaye donc de répondre à cette problématique.
  </p>
  <h1
    className="jsx-3052590751"
  >
    Propriétés des cartons et articles:
  </h1>
  <p
    className="jsx-3052590751"
  >
    Les articles à emballer sont de taille variable, représentée par un entier compris entre 1 et 9. Chaque carton a une capacité de contenance de 10. Ainsi, un carton peut par exemple contenir un article de taille 3, un article de taille 1, et un article de taille 6.
  </p>
  <JSXStyle
    css="section.jsx-3052590751{padding:0 15px;overflow-y:scroll;}h1.jsx-3052590751{margin:30px 0 20px 0;color:white;}"
    styleId="3052590751"
  />
</section>
`;
